regcore/models.py

```diff
@@ -1,21 +1,27 @@
 from django.db import models
 
+from mptt.models import MPTTModel, TreeForeignKey
+
 from regcore.fields import CompressedJSONField
 
 
-class Regulation(models.Model):
+class Regulation(MPTTModel):
+    id = models.TextField(primary_key=True)
+    parent = TreeForeignKey('self', null=True, blank=True, related_name='children', db_index=True)
     version = models.SlugField(max_length=20)
     label_string = models.SlugField(max_length=200)
     text = models.TextField()
     title = models.TextField(blank=True)
     node_type = models.SlugField(max_length=10)
-    children = CompressedJSONField()
     root = models.BooleanField(default=False, db_index=True)
 
     class Meta:
         index_together = (('version', 'label_string'),)
         unique_together = (('version', 'label_string'),)
 
+    class MPTTMeta:
+        order_insertion_by = ['label_string']
```

Comment:
I haven't looked too deeply into this, but I'm guessing this is how mptt orders the children of a node. Unfortunately, `label_string` is not a good mechanism for ordering. Consider:

* `111-2`, `111-11`
* `111-1-a-vi`, `111-1-a-ix`
* `111-1-p45`, `111-1-p123456784` (a definition)

In each case, a simple string comparison will fail.